codec-http/src/main/java/io/netty/handler/codec/http/cors/CorsHandler.java

```diff
@@ -80,7 +82,11 @@ private void handlePreflight(final ChannelHandlerContext ctx, final HttpRequest
             setPreflightHeaders(response);
         }
         release(request);
-        ctx.writeAndFlush(response).addListener(ChannelFutureListener.CLOSE);
+
+        final ChannelFuture future = ctx.writeAndFlush(response);
```

Comment:
If the protocol version is 1.0 and the request was marked ask keep-alive, then shouldn't the response be marked as keep-alive?  Do you need something like this here?

```java
if (HttpUtil.isKeepAlive(request)) {
  HttpUtil.setKeepAlive(response, true);
}
```